aws/rust-runtime/aws-inlineable/src/glacier_interceptors.rs

```diff
@@ -0,0 +1,454 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// This code is referenced in generated code, so the compiler doesn't realize it is used.
+#![allow(dead_code)]
+
+use aws_runtime::auth::sigv4::SigV4OperationSigningConfig;
+use aws_sig_auth::signer::SignableBody;
+use aws_smithy_http::body::SdkBody;
+use aws_smithy_http::byte_stream::{self, ByteStream};
+use aws_smithy_runtime_api::client::interceptors::context::phase::{
+    BeforeSerialization, BeforeTransmit,
+};
+use aws_smithy_runtime_api::client::interceptors::{BoxError, Interceptor, InterceptorContext};
+use aws_smithy_runtime_api::config_bag::ConfigBag;
+use bytes::Buf;
+use bytes_utils::SegmentedBuf;
+use http::header::{HeaderName, HeaderValue};
+use http::Request;
+use ring::digest::{Context, Digest, SHA256};
```

Comment:
Isn't there some AWS-owned crypto thing we can use for this? I remember people mentioning it when people were discussing FIPS compliance a few weeks ago.

Reply:
AWS LC? Definitely out of scope for this PR.